### ISSUE TITLE
Mail stuck in Outbox: TypeError: SendLaterFunction.waitAndDelete is not a function

### ISSUE THREAD

**Issue by padelt** [Extended-Thunder/send-later#47]
Just tried using the new version 8.0.9 and while sending still does not seem to work reliably (#16), it did once put a mail into the Outbox - but it is stuck there.
In the Logs (which I cranked up to "All" in the extension settings) I see:
```
WebExtensions: Saving message to C:\Users\pa\AppData\Local\Temp\tempMsg0.eml sl3u.js:86
WebExtensions: Failed to delete C:\Users\pa\AppData\Local\Temp\tempMsg0.eml. sl3u.js:430
TypeError: SendLaterFunctions.waitAndDelete is not a functionsl3u.js:431:38
    OnStopCopy jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:431
    CopyStringMessageToFolder jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:109
    sendRaw jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:447
    sendRaw self-hosted:844
    result resource://gre/modules/ExtensionParent.jsm:996
    withPendingBrowser resource://gre/modules/ExtensionParent.jsm:602
    result resource://gre/modules/ExtensionParent.jsm:996
    callAndLog resource://gre/modules/ExtensionParent.jsm:958
    recvAPICall resource://gre/modules/ExtensionParent.jsm:995
    AsyncFunctionNext self-hosted:693
```
Using the debugger I came to this place:
![grafik](https://user-images.githubusercontent.com/826969/97173206-77f1d900-1790-11eb-9aa1-97d37180bd80.png)
The exception is this:
```
Exception { name: "NS_ERROR_FILE_IS_LOCKED", message: "Component returned failure code: 0x8052000e (NS_ERROR_FILE_IS_LOCKED) [nsIFile.remove]", result: 2152857614, filename: "jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js", lineNumber: 428, columnNumber: 0, data: null, stack: "OnStopCopy@jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:428:27\nCopyStringMessageToFolder@jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:109:31\nsendRaw@jar:file:///D:/Thunderbird/zh7ot1a7.default/extensions/sendlater3@kamens.us.xpi!/experiments/sl3u.js:447:30\nrecvAPICall/result</<@resource://gre/modules/ExtensionParent.jsm:996:68\nwithPendingBrowser@resource://gre/modules/ExtensionParent.jsm:602:26\nrecvAPICall/result<@resource://gre/modules/ExtensionParent.jsm:996:24\ncallAndLog@resource://gre/modules/ExtensionParent.jsm:958:14\nrecvAPICall@resource://gre/modules/ExtensionParent.jsm:995:25\n", location: XPCWrappedNative_NoHelper }
```
Since the following line (`SendLaterFunctions.waitAndDelete(copying);`) fails, it seems like the mail getting stuck in the outbox is a consequence of that.

**Comment by padelt** [comment]
Small additional note: Once the mails are in the outbox and I manually trigger File -> Send Unsent Message, they are successfully sent.

**Comment by jperryhouts** [comment]
Thank you for catching that. That was exactly the problem. It was failing ungracefully on waitAndDelete (because of a typo in the actual function definition) and not continuing on to actually initiate the send operation. This should be fixed by #49 and available in the latest release package.
